workers.go

```diff
@@ -220,7 +224,7 @@ func (api *API) UploadWorker(ctx context.Context, rc *ResourceContainer, params
 		err         error
 	)
 
-	if params.Module || len(params.Bindings) > 0 {
+	if params.Module || params.Logpush != nil || len(params.Bindings) > 0 {
```

Comment:
could you expand on this logic for me? i'm not familiar enough with the usage of logpush jobs. are you unable to configure logpush for moduled based workers?

Reply:
Sure! Logpush can be enabled for both module & service worker formats, but the setting is set inside the metadata field which is only sent as part of the multipart body format here.

Reply:
this logic just doesn't feel right to me. if i'm reading this logic correctly and someone omits `params.Logpush` for a module based worker or has bindings, it won't ever end up calling `formatMultipartBody`, correct?

i think want we actually want here is a similar to https://github.com/cloudflare/cloudflare-go/blob/c071c00470ddeabd8cd9df4576cab11022be4622/workers.go#L270-L273 in the `formatMultipartBody` method where if it is defined, we want to append it to the struct.

```go
if params.Logpush != nil {
    meta.Logpush = params.Logpush
}
```

or am i missing something here?

Reply:
holiday brain and reading this as AND, not OR. this should be 👍